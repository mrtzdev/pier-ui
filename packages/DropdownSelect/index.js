import DropdownSelect from "./src/DropdownSelect";

export default DropdownSelect;
