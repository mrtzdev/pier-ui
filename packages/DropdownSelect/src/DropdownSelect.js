import PropTypes from "prop-types";

import DropdownSelectSingle from "./DropdownSelectSingle";
import DropdownSelectMultiple from "./DropdownSelectMultiple";

const propTypes = {
  //value: PropTypes.object.isRequired,
  options: PropTypes.array.isRequired,
  className: PropTypes.string,
  width: PropTypes.string,
  onChange: PropTypes.func,
  closeOnChange: PropTypes.bool,
  label: PropTypes.string,
  floatingLabel: PropTypes.bool,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  color: PropTypes.string,
  focusColor: PropTypes.string,
  size: PropTypes.oneOf(["base", "sm", "md", "lg"]),
  multiple: PropTypes.bool,
};

const defaultProps = {
  variant: "outline",
  size: "base",
  width: "auto",
  disabled: false,
  required: false,
  multiple: false,
};

const DropdownSelect = (props) => {
  const { multiple } = props;

  if (multiple && multiple !== undefined)
    return (
      <>
        <DropdownSelectMultiple {...props} />
      </>
    );
  else
    return (
      <>
        <DropdownSelectSingle {...props} />
      </>
    );
};

export default DropdownSelect;

DropdownSelect.propTypes = propTypes;
DropdownSelect.defaultProps = defaultProps;
