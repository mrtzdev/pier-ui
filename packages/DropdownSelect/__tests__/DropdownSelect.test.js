/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { axe, toHaveNoViolations } from "jest-axe";

import DropdownSelect from "../index";

expect.extend(toHaveNoViolations);

describe("Dropdown Select", () => {
  it("should render Input correctly", () => {
    render(
      <DropdownSelect
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );
    //screen.debug();
  });

  it("Renders with a wrapper className equal to the size", () => {
    const { container } = render(
      <DropdownSelect
        size="lg"
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );
    expect(container.firstChild).toHaveClass("wrap-lg");
  });

  it("Renders with className disabled", () => {
    const { container } = render(
      <DropdownSelect
        disabled
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );
    const dropdown = container.querySelector(".dropdown-select");
    expect(dropdown).toHaveClass("disabled");
  });

  it("Renders with label", () => {
    const { container } = render(
      <DropdownSelect
        label="testlabel"
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );
    const label = screen.getByText(/testLabel/i);

    expect(container).toContainElement(label);
  });

  it("calls onChange prop when clicked", () => {
    const handleChange = jest.fn();
    render(
      <DropdownSelect
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
        onChange={handleChange}
      />
    );
    const listItem = screen.getByText(/Mango/i);
    // screen.debug(listItem);
    fireEvent.click(listItem);
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("should render with class visible when clicked dropdown-title", () => {
    const { container } = render(
      <DropdownSelect
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );

    const dropdownTitle = container.querySelector(".dropdown-title");
    const dropdownContainer = container.querySelector(".dropdown-title");
    fireEvent.click(dropdownTitle);
    expect(dropdownTitle).toHaveClass("visible");
    expect(dropdownContainer).toHaveClass("visible");
  });

  it("should not have accessibility violations", async () => {
    const { container } = render(
      <DropdownSelect
        size="lg"
        options={[
          { label: "Grapes", value: "grapes" },
          { label: "Mango", value: "mango" },
          { label: "Strawberry", value: "strawberry" },
        ]}
        value={{ label: "Grapes", value: "grapes" }}
      />
    );

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
