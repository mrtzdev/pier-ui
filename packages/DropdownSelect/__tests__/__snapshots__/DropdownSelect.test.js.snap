// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Select should render DropdownSelect  correctly 1`] = `
<DropdownSelect
  disabled={false}
  multiple={false}
  options={
    Array [
      Object {
        "label": "Grapes",
        "value": "grapes",
      },
      Object {
        "label": "Mango",
        "value": "mango",
      },
      Object {
        "label": "Strawberry",
        "value": "strawberry",
      },
    ]
  }
  required={false}
  size="base"
  value={
    Object {
      "label": "Grapes",
      "value": "grapes",
    }
  }
  variant="outline"
  width="auto"
/>
`;
